### DISCUSSION TITLE
Limit characters in survey-text response box

### DISCUSSION THREAD

**Discussion by axifra** [jspsych/jsPsych#1763] [Q&A] [answered]
Hi all

At the beginning of our experiment we are using the survey-text plugin to ask participants to input an anonymous code they have been given (9 digits), as well as their age. However, we are having issues as some participants input the anonymous code twice. Is it possible to limit the number of characters of the response box in survey-text?

Thank you in advance!

**Comment by becky-gilbert** [comment, marked as answer]
Hi @axifra, yes you can do with a minor modification to the plugin file. You need to change the `maxlength` attribute in the `<input type="text">` tag. You may also want to change the `minlength` attribute to be the same number, if the code should always have exactly 9 digits (see [the docs about the input type="text" tag](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/text) for all of the options). [This is the relevant line](https://github.com/jspsych/jsPsych/blob/ab7ea845df8d12badc74cc338ebb1599e77efd41/plugins/jspsych-survey-text.js#L132) in the survey-text plugin. You could add the maxlength attribute anywhere in the input tag, like this:
```js
html += '<input type="text" maxlength="9" id="input-'+question_index+'"  name="#jspsych-survey-text-response-' + question_index + '" data-name="'+question.name+'" size="'+question.columns+'" '+autofocus+' '+req+' placeholder="'+question.placeholder+'"></input>';
```


Or if you don't want to change the plugin, you could use the [trial's on_load function](https://www.jspsych.org/overview/plugins/#the-on_load-parameter) to get the text input element from the page and [set its `maxlength` property](https://www.w3schools.com/jsref/prop_text_maxlength.asp):
```js
var enter_code = {
  type: 'survey-text',
  ... other parameters ...
  on_load: function() {
    // the ID in quotes below will depend on the index of your code question on the page,
    // first question is 'input-0', second is 'input-1', etc.
    document.getElementById('input-0').maxLength = 9;
  }
};
```

**Reply by axifra** [reply]
Hi @becky-gilbert, I used the on_load function and it worked wonderfully. Thank you so much for your help!
